docs/managing-compute-resources.asciidoc

```diff
@@ -0,0 +1,111 @@
+[id="{p}-managing-compute-resources"]
+== Managing compute resources
+
+When a Pod is created it may request CPU and RAM resources. It may also specify the maximum resources that the containers are allowed to consume. Both Pod `limits` and `requests` can be set in the specification of any object managed by the operator (Elasticsearch, Kibana or the APM server). For more information about how this is used by Kubernetes please see https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/[Managing Compute Resources for Containers].
+
+[float]
+[id="{p}-custom-resources"]
+=== Set custom resources
+
+The `resources` can be customized in the `podTemplate` of an object.
+
+Here is an example for Elasticsearch:
+
+[source,yaml]
+----
+spec:
+  nodes:
+  - podTemplate:
+      spec:
+        containers:
+        - name: elasticsearch
+          env:
+          - name: ES_JAVA_OPTS
+            value: -Xms2048M -Xmx2048M
+          resources:
+            requests:
+              memory: 4Gi
+              cpu: 1
+            limits:
```

Comment:
Nit: it might be helpful to set the limits to something different to make it clear they do not need to be the same as the request (though they can be)